test.js

```diff
@@ -0,0 +1,168 @@
+let check = 0;//check 0 edit and delete buttons set to save and cancel and check 1  save and cancel buttons set to edit and delete
+let deleted_row = [];
+let datalen;
+$(document).ready(function() {
+    console.log( "ready!");
+    $("button").click(function()
```

Comment:
use id to locate elements always. Moreover we should always try to remove the event listeners wherever possible whenever they are not needed.